Fix issue where default-repo wasn't being added to artifact tags (#5341)

dan-j opened the pull request:
<!-- Thank you for your contribution! -->

<!-- Include if applicable: -->
Fixes: #5341, #5463

**Description**
Using the `--default-repo` option on `deploy` commands wasn't being honoured and the default tag was being deployed.

It's a simple bug... The for-loop in `getBuildArtifactsAndSetTags` is correctly prefixing the tag with the --default-repo option but was setting it on the for-loop variable (which is a pointer re-used in each loop), and not updating the actual `buildArtifacts` slice.

I haven't added any tests, this is my first contribution and not sure how to get a `runner.Runner` in a test environment. If you want tests adding for this change could somebody give me a point in the right direction 🙂

dan-j commented on cmd/skaffold/app/cmd/util.go:
```diff
@@ -31,12 +31,12 @@ func getBuildArtifactsAndSetTags(r runner.Runner, artifacts []*latest.Artifact)
-	for _, artifact := range buildArtifacts {
-		tag, err := r.ApplyDefaultRepo(artifact.Tag)
+	for i := range buildArtifacts {
+		updatedTag, err := r.ApplyDefaultRepo(buildArtifacts[i].Tag)
```
renamed the variable because it was colliding with the `tag` package.

gsquared94 commented:
Thanks for your contribution @dan-j 
> I haven't added any tests, this is my first contribution and not sure how to get a `runner.Runner` in a test environment. If you want tests adding for this change could somebody give me a point in the right direction 🙂

You can modify the function signature to:
```go
func getBuildArtifactsAndSetTags(artifacts []*latest.Artifact, applyDefaultRepo func(string)(string, error)) ([]build.Artifact, error) {
  ...
```
and all non-test references look like:
```go
buildArtifacts, err := getBuildArtifactsAndSetTags(artifacts, r.ApplyDefaultRepo)
```

That removes the `pkg/skaffold/runner` dependency from this file.

dan-j commented:
Oops, I had forgotten about this and then just ran into the problem this solves 😬

I got halfway through the tests but then sidetracked, will finish this off this week at some point

MarlonGamez commented:
hey @dan-j, have you had any time to work on this? Not meaning to pressure, just want to make sure that you have bandwith to finish this :) If not, I'm sure someone on the team could pick it up

dan-j commented:
Hey! Yeah I got completely sidetracked by other stuff and forgot, until I've just hit the exact same issue again when running a manual deployment 😅

I was already halfway through the tests, will try finish them tonight, if not it will be this week.

dan-j commented:
OK, I've added some tests.

The change @gsquared94 suggested wasn't completely straight forward because that function also uses the global flags for merging build artifacts. I've refactored the problematic code into a new function and have added tests for this new function.

MarlonGamez reviewed (approved):
looks good to me! Thanks @dan-j for this :)
